frontend-ssr/pages/submissions.tsx

```diff
@@ -22,22 +22,49 @@ interface State {
 }
 
 class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private hiddenRef = React.createRef<HTMLDivElement>();
+
   constructor(props) {
     super(props);
 
     this.state = {
-      loading: props.submissions.submissions === undefined,
+      loading: false,
     };
   }
 
   componentDidMount() {
-    let { submissions } = this.props;
+    this.initIntersectionObserver();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { submissions } = this.props;
 
-    if (submissions.submissions === undefined) {
-      this.getData();
+    if (submissions !== prevProps.submissions && submissions.end) {
+      this.observer.disconnect();
     }
   }
 
+  componentWillUnmount() {
+    this.observer.disconnect();
+  }
+
+  initIntersectionObserver = () => {
+    const options = {
+      threshold: 0.3,
+    };
+
+    this.observer = new IntersectionObserver(this.loadMore, options);
+
+    const elToObserve = this.hiddenRef.current;
```

Comment:
We can delete all the code below and replace directly with `this.observer.observe(this.hiddenRef.current);` since we know the hiddenRef will always be there?

Reply:
yep. we know :)